x-pack/plugins/security_solution/common/endpoint/index_data.ts

```diff
@@ -98,3 +198,262 @@ async function indexAlerts(
     await client.bulk({ body, refresh: true });
   }
 }
+
```

Comment:
@paul-tavares this is a significant code change to the core class. I think it may be good to separate this code out. And from a cursory look it may be difficult to maintain and change. Perhaps a follow up PR with that view in mind. Because if there is any changes in the system we have to change this code also.

Reply:
@nnamdifrankie Agreed - some of this needs to be broken out into smaller set of classes/namespaces. but: could you clarify what class you are referring to? (or did you mean the entire module `index_data.ts`?) 

Also agree that if the system changes, this will have to be updated, but I think it valuable to have the ability to create a fake set of data across the entire system (minus the actual endpoint). Lets disucss if perhaps you feel strongly that we should not be creating this type of "full feature" tooling 😄 

also - just FYI: my idea is to ultimately create a `FakeAgent` class that will allows us to enroll and control agents against fleet without actually having an agent running (just sending messages to fleet - [some of that will land in this external tool](https://github.com/paul-tavares/artifact-downloader)). The ability to do this will significantly improve our ability to do CI end-to-end testing within Kibana (Functional testing) - especially for our Endpoint (hosts) List page which is driven by a combination of both endpoint metadata and Fleet/Agent data _(maybe even create a FakeEndpoint class to mimic Endpoints sending data to elastic)_.